packages/flutter_tools/lib/src/http_host_availability_validator.dart

```diff
@@ -0,0 +1,123 @@
+// @dart = 2.9
+import 'base/io.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+
+/// Hosts used by flutter on all machines
+const List<String> commonRequiredHostUrls = <String>[
+  'https://cloud.google.com/',
+  'https://maven.google.com/',
+  'https://pub.dev/',
+];
+
+/// Hosts used only on MacOS
+const List<String> macOsRequiredHostUrls = <String>[
+  'https://cocoapods.org/',
+];
+
+/// Validation class that checks if all the given URLs are reachable
+class HttpHostAvailabilityValidator extends DoctorValidator {
+  HttpHostAvailabilityValidator({
+    Platform platform,
+    HttpClient httpClient,
+  })   : _platform = platform,
+        _httpClient = httpClient,
+        super('HTTP host availability');
+
+  final Platform _platform;
+  final HttpClient _httpClient;
+
+  @override
+  String get slowWarning => 'HTTP host availability check is taking a long time';
+
+  /// Returns a list of URLs to check availability, different for different platforms
+  List<String> get _allRequiredHosts {
```

Comment:
Technically both the pub and cloud URL can be overriden with environment variables,  see https://flutter.dev/community/china 

Only Android applications need access to maven, it might be worth excluding from this check

Reply:
Thanks for that, I had not thought of the environment variable overrides. I added it in 070ae59 to the validator, but did not know how to access environment variables while writing tests. Could you please guide me on that?

Reply:
Also, I kept the maven check, as the original issue (#49403, which references #22470 and #30597) was regarding maven not being available.

Reply:
Environment variables are present on the `Platform` class, you can inject this into the interface and use a FakePlatform for testing

Reply:
Thanks, I'll add it.